dev-lang/zig/zig-0.13.0-r3.ebuild

```diff
@@ -0,0 +1,355 @@
+		fi
+	else
+		cd "${BUILD_DIR}" || die
+		ln -s "${S}/stage1/" . || die
+		ln -s "${S}/src/" . || die
+		ln -s "${S}/lib/" . || die
+
+		local native_cc="$(tc-getBUILD_CC)"
+		"${native_cc}" -o bootstrap "${S}/bootstrap.c" || die "Zig's bootstrap.c compilation failed"
+		ZIG_HOST_TARGET_TRIPLE="${ZIG_HOST_AS_TARGET}" CC="${native_cc}" edo ./bootstrap
+	fi
+
+	cd "${BUILD_DIR}" || die
+	ZIG_EXE="./zig2" zig_src_compile --prefix stage3/
+
+	# Requires running stage3 which is built for cross-target.
+	if ! tc-is-cross-compiler; then
+		./stage3/bin/zig env || die "Zig compilation failed"
+
+		if use doc; then
+			ZIG_EXE="./stage3/bin/zig" zig_src_compile langref --prefix "${S}/docgen/"
+		fi
+	fi
+}
+
+src_test() {
+	if has_version -b app-emulation/qemu; then
+		ewarn "QEMU executable was found on your building system."
+		ewarn "If you have qemu-binfmt (binfmt_misc) hooks enabled for"
+		ewarn "foreign architectures, Zig tests might fail."
+		ewarn "In this case, please disable qemu-binfmt and try again."
+	fi
+
+	cd "${BUILD_DIR}" || die
+
+	# XXX: When we pass a libc installation to Zig, it will fail to find
+	#      the bundled libraries for targets like aarch64-macos and
+	#      *-linux-musl. Zig doesn't run binaries for these targets when
+	#      -Dskip-non-native is passed, but they are still compiled, so
+	#      the test will fail. There's no way to disable --libc once passed,
+	#      so we need to strip it from ZBS_ARGS.
+	#      See: https://github.com/ziglang/zig/issues/22383
+	local args_backup=("${ZBS_ARGS[@]}")
+
+	for ((i = 0; i < ${#ZBS_ARGS[@]}; i++)); do
+		if [[ "${ZBS_ARGS[i]}" == "--libc" ]]; then
+			unset ZBS_ARGS[i]
+			unset ZBS_ARGS[i+1]
+			break
+		fi
+	done
+
+	ZIG_EXE="./stage3/bin/zig" zig_src_test -Dskip-non-native
+
+	ZBS_ARGS=("${args_backup[@]}")
+}
+
+src_install() {
+	use doc && local HTML_DOCS=( "docgen/doc/langref.html" )
+
+	ZIG_EXE="./zig2" zig_src_install --prefix "${ZIG_SYS_INSTALL_DEST}"
+
+	cd "${D}/${ZIG_SYS_INSTALL_DEST}" || die
+	mv lib/zig/ lib2/ || die
+	rm -rf lib/ || die
+	mv lib2/ lib/ || die
+	dosym -r "${ZIG_SYS_INSTALL_DEST}/bin/zig" /usr/bin/zig-${PV}
+}
+
+pkg_postinst() {
+	eselect zig update ifunset || die
+
+	elog "Starting from 0.12.0, Zig no longer installs"
+	elog "precompiled standard library documentation."
+	elog "Instead, you can call \`zig std\` to compile it on-the-fly."
+	elog "It reflects all edits in standard library automatically."
+	elog "See \`zig std --help\` for more information."
+	elog "More details here: https://ziglang.org/download/0.12.0/release-notes.html#Redesign-How-Autodoc-Works"
+
+	if ! use llvm; then
+		elog "Currently, Zig built without LLVM support lacks some"
+		elog "important features such as most optimizations, @cImport, etc."
+		elog "They are listed under \"Building from Source without LLVM\""
+		elog "section of the README file from \"/usr/share/doc/${PF}\" ."
+	fi
+}
+
+pkg_postrm() {
+	eselect zig update ifunset || die
```

Comment:
Ah sorry, I think I messed up this when doing it separately from https://github.com/gentoo/gentoo/pull/40215, copied fix there https://github.com/gentoo/gentoo/pull/40598